core/src/main/scala/kafka/log/streamaspect/ElasticLog.scala

```diff
@@ -634,6 +636,7 @@ object ElasticLog extends Logging {
                 stream
             } else {
                 val metaStreamId = Unpooled.wrappedBuffer(value.get()).readLong()
+                awaitStreamReadyForOpen(openStreamChecker, metaStreamId, leaderEpoch, logIdent = logIdent)
```

Comment:
Is that OK to block the current thread here?

Reply:
> Is that OK to block the current thread here?

The Open/Close Partition process is entirely blocked. Opening/closing involves many stages, and the main reason for this approach is to simplify the coding.

To improve the overall concurrency, there is a CachedThreadPool with maxSize=256 in ElasticReplicaManager.